airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java

```diff
@@ -188,30 +188,35 @@ public static void setStatementField(PreparedStatement preparedStatement,
       case FLOAT, DOUBLE -> preparedStatement.setDouble(parameterIndex, Double.parseDouble(value));
       case REAL -> preparedStatement.setFloat(parameterIndex, Float.parseFloat(value));
       case NUMERIC, DECIMAL -> preparedStatement.setBigDecimal(parameterIndex, new BigDecimal(value));
-      case CHAR, VARCHAR, LONGVARCHAR -> preparedStatement.setString(parameterIndex, value);
+      case CHAR, NCHAR, NVARCHAR, VARCHAR, LONGVARCHAR -> preparedStatement.setString(parameterIndex, value);
       case BINARY -> preparedStatement.setBytes(parameterIndex, DatatypeConverter.parseHexBinary(value));
+      // since cursor are expected to be comparable, handle cursor typing strictly and error on
+      // unrecognized types
       default -> throw new IllegalArgumentException(String.format("%s is not supported.", cursorFieldType));
     }
   }
 
   // the switch statement intentionally has duplicates so that its structure matches the type switch
   // statement above.
-
+  // these json type fields are eventually consumed by the normalization process (if configured).
```

Comment:
Does something need to change in normalization to support the new primitive type `INTEGER`?

Reply:
No, other sources were already producing `integer` columns as json primitives

Reply:
i think more broadly the question is just what needs to change downstream to support the integer type?

Reply:
this may be a little puritanical of me, but it seems a little odd to have a comment about normalization in this utils class. feel free to keep it if you feel it's helpful though. (2 of 10 on the scale)

Reply:
In normalization, it is already handling `integer` fields.

For example facebook catalog produces such as this one: https://github.com/airbytehq/airbyte/blob/f5094b558ad77a5985a54bd76c2cfca9a891a027/airbyte-integrations/bases/base-normalization/unit_tests/resources/facebook_catalog.json#L104